src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -0,0 +1,222 @@
+                            SqlConnection sqlConn = GetSqlConnection(connInfo);
+                            fileValidator = new DisasterRecoveryFileValidator(sqlConn,
+                                fileBrowserParams.ServiceType == ServiceConstants.Restore);
+                            break;
+                        default:
+                            break;
+                    }
+
+                    if (fileValidator != null)
+                    {
+                        string errorMessage = "";
+                        fileValidator.ValidatePaths(fileBrowserParams.SelectedFiles, out errorMessage);
+                        if (!string.IsNullOrEmpty(errorMessage))
+                        {
+                            response.Succeeded = false;
+                            response.Message = errorMessage;
+                        }
+                        else
+                        {
+                            response.Succeeded = true;
+                        }
+                    }
+                    else
+                    {
+                        response.Succeeded = true;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                response.Succeeded = false;
+                response.Message = ex.Message;
+            }
+
+            await requestContext.SendResult(response);
+        }
+
+        internal async Task HandleFileBrowserCloseRequest(
+            FileBrowserCloseParams fileBrowserParams,
+            RequestContext<FileBrowserCloseResponse> requestContext)
+        {
+            FileBrowserCloseResponse response = new FileBrowserCloseResponse();
+            if (this.ownerToFileBrowserMap.ContainsKey(fileBrowserParams.OwnerUri))
+            {
+                this.ownerToFileBrowserMap.Remove(fileBrowserParams.OwnerUri);
+                response.Succeeded = true;
+            }
+            await requestContext.SendResult(response);
+        }
+
+        internal static SqlConnection GetSqlConnection(ConnectionInfo connInfo)
+        {
+            try
+            {
+                // increase the connection timeout to at least 30 seconds and and build connection string
+                // enable PersistSecurityInfo to handle issues in SMO where the connection context is lost in reconnections
+                int? originalTimeout = connInfo.ConnectionDetails.ConnectTimeout;
+                bool? originalPersistSecurityInfo = connInfo.ConnectionDetails.PersistSecurityInfo;
+                connInfo.ConnectionDetails.ConnectTimeout = Math.Max(30, originalTimeout ?? 0);
+                connInfo.ConnectionDetails.PersistSecurityInfo = true;
+                string connectionString = ConnectionService.BuildConnectionString(connInfo.ConnectionDetails);
+                connInfo.ConnectionDetails.ConnectTimeout = originalTimeout;
+                connInfo.ConnectionDetails.PersistSecurityInfo = originalPersistSecurityInfo;
```

Comment:
I've seen this snippet elsewhere in the code..we should move to a central place to avoid spreading this logic too many places?

also, does file browser really needs its own connection or can't it get once from its parent operation..backup in this case?

Reply:
no, the file browser doesn't need its own connection after my fix on backup connection. I updated it to get existing connection here.